flink-ml-parent/flink-ml-api/src/main/java/org/apache/flink/ml/api/misc/param/Params.java

```diff
@@ -47,14 +70,33 @@
 	 * @throws RuntimeException if the Params doesn't contains the specific parameter, while the
 	 *                          param is not optional but has no default value in the {@code info}
 	 */
-	@SuppressWarnings("unchecked")
 	public <V> V get(ParamInfo<V> info) {
-		V value = (V) paramMap.getOrDefault(info.getName(), info.getDefaultValue());
-		if (value == null && !info.isOptional() && !info.hasDefaultValue()) {
-			throw new RuntimeException(info.getName() +
-				" not exist which is not optional and don't have a default value");
+		Stream<V> stream = getParamNameAndAlias(info)
+			.filter(this.params::containsKey)
+			.map(x -> this.params.get(x))
+			.map(x -> valueFromJson(x, info.getValueClass()))
+			.limit(1);
```

Comment:
Do you need the stream object here?
How do you feel about `findFirst()` method here?

Reply:
If not use stream object, we need more codes. 

For the value of parameter may be null, and the findFirst method is not friendly for the null, thus we chose the limit(.) method.

When the user sets a Null as the Value of Param, using the findFirst method in the Stream class will throw NullPointerException.
Looking deep into the code, findFirst returns an instance of Optional<T>,
but Optional<T> can only use non-null value as a constructor argument.
For example, the following code will throw NullPointerException.
<code>
    Stream.of(null, "2").findFirst();
</code>

Reply:
paramValue here has only one element. I think it's better to replace with the only element and it will make the code below much easier.

Reply:
> If not use stream object, we need more codes.
> 
> For the value of parameter may be null, and the findFirst method is not friendly for the null, thus we chose the limit(.) method.
> 
> When the user sets a Null as the Value of Param, using the findFirst method in the Stream class will throw NullPointerException.
> Looking deep into the code, findFirst returns an instance of Optional,
> but Optional can only use non-null value as a constructor argument.
> For example, the following code will throw NullPointerException.
> ` Stream.of(null, "2").findFirst(); `

I suppose if we will have `null` in the stream is an invalid state, and we need to handle this situation.
anyway, we can filtering `null` before getting results, for e.g. by `.filter(Objects::nonNull)`

I agree with @becketqin  and @c4emmmm that the code without stream will be much easier to reading

Reply:
Thanks for comments and code snippet. The implementation without Stream is better. For the case of specified values for multiple alias, I think it is better to throw exception.